executor/src/witgen/jit/compiler.rs

```diff
@@ -29,10 +29,15 @@ use super::{
     variable::Variable,
 };
 
-pub struct WitgenFunction<T> {
+pub enum WitgenFunction<T: FieldElement> {
```

Comment:
I think it would be better to keep `compiler.rs` only concerned about compiling effects and transforming parameters to fit the loaded function.

What do you think about putting the enum in function_cache? Then we also don't need a `#[cfg(test)]`-function for the unit tests here.